app/lib/actions/katello/pulp_selector.rb

```diff
@@ -2,12 +2,31 @@ module Actions
   module Katello
     class PulpSelector < Actions::Base
       def plan(backend_actions, repository, smart_proxy, *args)
+        pass_only_args = args.dig(0, :args_only) || false
+        return_action_output = args.dig(0, :return_output) || false
         backend_type = smart_proxy.backend_service_type(repository)
         found_action = backend_actions.find { |action| action.backend_service_type == backend_type }
 
         fail "Could not locate an action for type #{backend_type}" unless found_action
-        plan_action(found_action, repository, smart_proxy, *args)
+        if return_action_output
+          sequence do
+            if pass_only_args
+              input[:output] = plan_action(found_action, *args).output
+              plan_self(input)
+            else
+              input[:output] = plan_action(found_action, repository, smart_proxy, *args).output
+              plan_self(input)
+            end
+          end
+        else
+          planned_action = pass_only_args ? plan_action(found_action, *args) : plan_action(found_action, repository, smart_proxy, *args)
+        end
+      end
+
+      def run
+        output[:propogated] = input[:output]
```

Comment:
I wonder if we could just do something like:
output = input[:output] ?  

or maybe, if we have to, something like:
```
input[:output].each do |key, value|
  output[key] = value
end
```
This way, we can treat the pulp_selector just like we do the original action